src/rest.cpp

```diff
@@ -355,13 +367,30 @@ static bool rest_filter_header(const std::any& context, HTTPRequest* req, const
 
     std::vector<std::string> uri_parts;
     boost::split(uri_parts, param, boost::is_any_of("/"));
-    if (uri_parts.size() != 3) {
-        return RESTERR(req, HTTP_BAD_REQUEST, "Invalid URI format. Expected /rest/blockfilterheaders/<filtertype>/<count>/<blockhash>");
+    std::string raw_count {};
+    size_t blockhash_index {};
+    if (uri_parts.size() == 3) {
+        // deprecated path: /rest/blockfilterheaders/<filtertype>/<count>/<blockhash>
+        blockhash_index = 2;
+        raw_count = uri_parts[1];
+    }
+    else if (uri_parts.size() == 2) {
+        // new path with query parameter: /rest/blockfilterheaders/<filtertype>/<blockhash>?count=<count>
+        blockhash_index = 1;
+        raw_count = req->GetQueryParameter("count", "5");
+    }
+    else {
+        return RESTERR(req, HTTP_BAD_REQUEST, "Invalid URI format. Expected /rest/blockfilterheaders/<filtertype>/<blockhash>.<ext>?count=<count>");
+    }
+
+    auto parsed_count = ToIntegral<size_t>(raw_count);
```

Comment:
Any reason to lose the `const` here?

Reply:
Good catch, this is leftover from before I introduced the new `raw_count` variable but indeed should be `const` now. Fixed in both `rest_headers` and `rest_filter_header`.